duckduckgo/macos-browser
# Replace altool with notarytool in release build script

Task/Issue URL: https://app.asana.com/0/0/1202424653108929/f
CC: @samsymons 

**Description**:
* I replaced altool with notarytool in the build scripts. It allowed to simplify the script slightly as
  notarytool supports awaiting notarization itself.
* For CI builds, I switched to using App Store Connect API Key for notarytool. Command-line builds
  still use Apple ID and application-specific password because it was easier (faster) to integrate within
  the current script (otherwise the user would have to provide the API key ID, private key file and
  the issuer UUID every time they wanted to run the script). After all, the script is mostly used in the CI anyway.
* I fixed the "Create DMG" flag so that when it's off, the DMG is not created. This wasn't the case until now.



**Steps to test this PR**:
1. Go to Actions and trigger "Make Release Build" from `dominik/notarytool` branch. Uncheck "Create DMG image".
1. Verify that the run passes and it outputs a working notarized application, and the DMG is not created.
1. Run another build, but check "Create DMG image".
1. Verify that the DMG is created.

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
